File: src/main/java/java_examples/2_immutability.java
package java_examples;

import java.util.HashMap;
import java.util.Map;

class Immutability {

    public static void main(String[] args) {
        // In Java, to adjust an entry in a map we can simply adjust it in-place as it's mutable

        Map<String, String> original_map = new HashMap<>();
        original_map.put("one", "value");
        original_map.put("another", "entry");

        System.out.println(original_map);

        original_map.put("one", "some_other_value");

        System.out.println(original_map);

    }
}

File: src/main/scala/scala_examples/2_immutability.scala
package scala_examples

object Immutability extends App {
  // In Scala, to adjust an entry in a map we copy the previous along with the altered entry to create a new object
  // as it's immutable

  val original_map = Map(
    "one" -> "value",
    "another" -> "entry"
  )
  println(original_map)

  val new_map = original_map + ("one" -> "some_other_value")
  println(new_map)
}
